src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/IO/TdsReadStream.cs

```diff
@@ -0,0 +1,340 @@
+            await PrepareBufferIfNeeded(isAsync, ct).ConfigureAwait(false);
+            return _readBuffer[_readIndex];
+        }
+
+        /// <inheritdoc />
+        public override int Read(Span<byte> buffer)
+        {
+            int lengthToFill = buffer.Length;
+            int totalRead = 0;
+            while (lengthToFill > 0)
+            {
+                PrepareBufferIfNeeded(isAsync: false, CancellationToken.None).ConfigureAwait(false);
+
+                int lengthToCopy = MinDataAvailableBeforeRead(lengthToFill);
+                ReadOnlySpan<byte> copyFrom = _readBuffer.AsSpan(_readIndex, lengthToCopy);
+                copyFrom.CopyTo(buffer.Slice(totalRead, lengthToFill));
+                totalRead += lengthToCopy;
+                lengthToFill -= lengthToCopy;
+                AdvanceBufferOnRead(lengthToCopy);
+            }
+            return totalRead;
+        }
+
+        /// <inheritdoc />
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return Read(buffer.AsSpan(offset, count));
+        }
+
+        /// <inheritdoc />
+        public override async ValueTask<int> ReadAsync(Memory<byte> buffer, CancellationToken cancellationToken)
+        {
+            int lengthToFill = buffer.Length;
+            int totalRead = 0;
+            while (lengthToFill > 0)
+            {
+                await PrepareBufferIfNeeded(isAsync: true, cancellationToken).ConfigureAwait(false);
+                int lengthToCopy = MinDataAvailableBeforeRead(lengthToFill);
+                ReadOnlyMemory<byte> copyFrom = new ReadOnlyMemory<byte>(_readBuffer, _readIndex, lengthToCopy);
+                copyFrom.CopyTo(buffer.Slice(totalRead, lengthToFill));
+                totalRead += lengthToCopy;
+                lengthToFill -= lengthToCopy;
+                AdvanceBufferOnRead(lengthToCopy);
+            }
+            return totalRead;
+        }
+
+        /// <inheritdoc />
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            return await ReadAsync(buffer.AsMemory(offset, count), cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc />
+        public void ReplaceUnderlyingStream(Stream stream) => _underlyingStream = stream;
+
+        /// <inheritdoc />
+        public override long Seek(long offset, SeekOrigin origin) => throw new NotSupportedException();
+
+        /// <inheritdoc />
+        public override void SetLength(long value) => throw new NotSupportedException();
+
+        /// <inheritdoc />
+        public void SetPacketSize(int bufferSize)
+        {
+            _readBuffer = new byte[bufferSize];
+        }
+
+        /// <inheritdoc />
+        public async ValueTask SkipReadBytesAsync(int skipCount, bool isAsync, CancellationToken ct)
+        {
+            int lengthLeftToSkip = skipCount;
+            int totalRead = 0;
```

Comment:
This variable is being updated but its never used after being updated. So it can probabaly be safely removed

Reply:
yes